### PR TITLE
Docs: unified GC and deprecate old GC

### LINKED ISSUE #6223: Document unified GC: explain deprecation plan and new way to run

### PR DESCRIPTION
**Pull request by johnnyaug**
Resolves #6223.

### CODE REVIEW

File: docs/howto/garbage-collection.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+{
+  "default_retention_days": 14,
+  "branches": [
+    {"branch_id": "main", "retention_days": 21},
+    {"branch_id": "dev", "retention_days": 7}
+  ]
+}
+```
+
+In the above example, objects are retained for 14 days after deletion by default. However, if they are present in the branch `main`, they are retained for 21 days.
```
Comment (N-o-Z) [inline docs/howto/garbage-collection.md:51]:
```suggestion
In the above example, objects are retained for 14 days after deletion by default. However, if they are present in the branch `main`, they will be retained for 21 days.
```

File: docs/howto/garbage-collection.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+Objects present in the `dev` branch (but not in any other branch) are retained for 7 days after they are deleted.
```
Comment (N-o-Z) [inline docs/howto/garbage-collection.md:52]: after they are deleted?

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Looks good, some comments.
A general side note: Might want to consider changing the "you/your" terminology

File: docs/howto/garbage-collection.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+{
+  "default_retention_days": 14,
+  "branches": [
+    {"branch_id": "main", "retention_days": 21},
+    {"branch_id": "dev", "retention_days": 7}
+  ]
+}
+```
+
+In the above example, objects will be retained for 14 days after deletion by default.
```
Comment (Jonathan-Rosenberg) [inline docs/howto/garbage-collection.md:51]: I wonder if we should add the overridden scenario, i.e. paths that were overridden and therefore the objects that previously represented them were deleted

File: docs/howto/garbage-collection.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+```bash
+lakectl gc set-config lakefs://example-repo -f example_repo_gc_rules.json 
+```
+
+</div>
+<div markdown="1" id="ui-option">
+From the lakeFS web UI:
+
+1. Navigate to the main page of your repository.
+2. Go to _Settings_ -> _Retention_.
```
Comment (Jonathan-Rosenberg) [inline docs/howto/garbage-collection.md:90]:
```suggestion
2. Go to _Settings_ -> _Garbage Collection_.
```
!

File: docs/howto/garbage-collection.md [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+  ```
+  </div>
+
+  <div markdown="1" id="azure-option">
+
+If you want to access your storage using the account key:
+
+  ```bash
+spark-submit --class io.treeverse.gc.GarbageCollection \
+  --packages org.apache.hadoop:hadoop-aws:3.2.1 \
```
Comment (Jonathan-Rosenberg) [inline docs/howto/garbage-collection.md:142]:
```suggestion
  --packages org.apache.hadoop:hadoop-azure:3.2.1 \
```
no?

File: docs/howto/garbage-collection.md [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+  -c spark.hadoop.fs.azure.account.key.<AZURE_STORAGE_ACCOUNT>.dfs.core.windows.net=<AZURE_STORAGE_ACCESS_KEY> \
+  http://treeverse-clients-us-east.s3-website-us-east-1.amazonaws.com/lakefs-spark-client-312-hadoop3/0.8.1/lakefs-spark-client-312-hadoop3-assembly-0.8.1.jar \
+  example-repo
+  ```
+
+Or, if you want to access your storage using an Azure service principal:
+
+  ```bash
+spark-submit --class io.treeverse.gc.GarbageCollection \
+  --packages org.apache.hadoop:hadoop-aws:3.2.1 \
```
Comment (Jonathan-Rosenberg) [inline docs/howto/garbage-collection.md:155]: same here

**Review by Jonathan-Rosenberg — CHANGES_REQUESTED** [review]
Great stuff.
The requested change is for line [90](https://github.com/treeverse/lakeFS/pull/6228/files#diff-4866326515d944e9160cd7490279ef82f0e875e79f57b293ea9bb999ab3f458eR90)

File: docs/howto/garbage-collection.md (hunk 3 above)
Comment (johnnyaug) [reply docs/howto/garbage-collection.md:53]: Added

File: docs/howto/garbage-collection.md (hunk 4 above)
Comment (johnnyaug) [reply docs/howto/garbage-collection.md]: Done

**Review by N-o-Z — APPROVED** [review]
LGTM!

File: docs/howto/garbage-collection.md (hunk 5 above)
Comment (Jonathan-Rosenberg) [reply docs/howto/garbage-collection.md:144]: Is `hadoop-aws` needed here?
I know it's not your addition, yet it might be redundant...

**Review by Jonathan-Rosenberg — APPROVED** [review]
approving with the `hadoop-aws` concern in progress

**Pull request merged**